src/vnsw/agent/services/icmpv6_handler.cc

```diff
@@ -377,11 +354,11 @@ void Icmpv6Handler::SolicitedMulticastIpAndMac(const Ip6Address &dip,
     /* The ethernet address for IPv6 multicast address is 0x33-33-mm-mm-mm-mm,
      * where mm-mm-mm-mm is a direct mapping of the last 32 bits of the
      * IPv6 multicast address */
-    mac[(size_t)0] = mac[(size_t)1] = 0x33;
-    mac[(size_t)2] = ip[12];
-    mac[(size_t)3] = ip[13];
-    mac[(size_t)4] = ip[14];
-    mac[(size_t)5] = ip[15];
+    mac[0] = mac[1] = 0x33;
```

Comment:
BTW, I don't understand why it was `mac[(size_t)0]`(etc) here in the first place?